src/ReplicatedStorage/Client/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,63 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local ContextActionService = game:GetService("ContextActionService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local clientComm = Comm.ClientComm.new(character, false, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._comm = clientComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
+	self._serverObject = clientComm:BuildObject()
+	self._owner = Players:GetPlayerFromCharacter(character)
+	self.Instance = character
+
+	self._trove:Connect(character.AncestryChanged, function()
+		self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	end)
+	if not self:IsOwner() then
+		return self
+	end
```

Comment:
This is slightly wild - but I'll allow it.

Reply:
Also related to me. It's in the jetpack code as well.